synapse/appservice/__init__.py

```diff
@@ -316,7 +316,7 @@ async def is_interested_in_presence(
 
         # Then find out if the appservice is interested in any of those rooms
         for room_id in room_ids:
-            if await self.matches_user_in_member_list(
+            if await self.is_interested_in_room(
```

Comment:
Sorry -- I'm not following this change, it seems bigger than just "make `matches_user_in_member_list` private"

I think the change is OK from the rest of the commit message, but I'm having trouble following it.

Reply:
Ah, yes it is. `is_interested_in_room` calls `_matches_user_in_member_list`, but it will also return True if:

* the room's ID is part of the application service's room ID namespace
* one of the room's aliases are part of the application service's room alias namespace

which is the intended functionality according to [the spec](https://spec.matrix.org/v1.2/application-service-api/#registration):

> Application services register “namespaces” of user IDs, room aliases and room IDs. These namespaces are represented as regular expressions. An application service is said to be “interested” in a given event if one of the IDs in the event match the regular expression provided by the application service, such as the room having an alias or ID in the relevant namespaces. Similarly, the application service is said to be interested in a given event if one of the application service’s namespaced users is the target of the event, or is a joined member of the room where the event occurred.